### PR TITLE
Updates from rspec-dev (2017-11-13)

### PR DESCRIPTION
**Pull request by JonRowe**
These are some updates, generated from rspec-dev's rake tasks.

### CODE REVIEW

**Comment by myronmarston** [comment]
We've been able to run our cukes on JRuby in the past.  Here's a build from about a month ago:

https://travis-ci.org/rspec/rspec-core/builds/289094713

The JRuby parts of the build passed in 12-22 minutes.  Do we know what changed and why it takes so much longer now?

**Comment by JonRowe** [comment]
I think I removed the cache as part of debugging the other JRuby build issue, which may be it, but given that it's consistently failing now I think it's a good idea to stop running it. JRuby has always been slower and we haven't really had many issues with cucumber failing specifically on JRuby that we need to catch, and we should move any such regressions into the test suite itself as we already have confusion from people about that line between documentation and test.

File: script/custom_build_functions.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  if [ -d features ]; then
+    echo "${PWD}/bin/cucumber"
+
+    if is_mri_192; then
+      # For some reason we get SystemStackError on 1.9.2 when using
+      # the bin/cucumber approach below. That approach is faster
+      # (as it avoids the bundler tax), so we use it on rubies where we can.
+      bundle exec cucumber --strict
+    elif is_jruby; then
+      return 0
```
Comment (myronmarston) [inline script/custom_build_functions.sh]: IMO, we shouldn't silently skip a major part of the build.  It should print a message in this branch so it is clear why the cukes didn't run based just on looking at the travis build output.

Comment (JonRowe) [reply script/custom_build_functions.sh]: I'm echoing a message now

**Comment by myronmarston** [comment]
LGTM!

**Pull request merged**